### PR TITLE
Review checks fixes

### PR DESCRIPTION
**Pull request by jameesjohn**
## Explanation
Fixes oppiabot adding the LGTM label early. Oppiabot now adds the label after all reviewers have approved the pull request.
Fixes oppiabot periodic check showing that a PR needs help from onboarding team. This was caused by a bug which checked the wrong status from the GitHub API.

## Checklist
- [ ] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [ ] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

File: lib/checkPullRequestReview.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+const { context } = require('@actions/github');
```
Comment (ankita240796) [inline lib/checkPullRequestReview.js:1]: Is this required? If yes, why wasn't it present before?

File: lib/utils.js [hunk 2]
```diff
@@ -162,7 +162,7 @@ const hasPullRequestBeenApproved = async (context, pullRequestNumber) => {
     })
   );
 
-  return searchResult.status === SUCCESS_STATUS;
+  return searchResult.data.items.length > 0;
```
Comment (ankita240796) [inline lib/utils.js:165]: Explain with a comment why we are checking like this (same as what we discussed offline.)

File: lib/utils.js [hunk 3]
```diff
@@ -188,7 +188,7 @@ const doesPullRequestHaveChangesRequested = async (
...
```
Comment (ankita240796) [inline lib/utils.js:191]: Diito.

File: lib/utils.js (hunk 2 above)
Comment (ankita240796) [reply lib/utils.js:165]: Also, if the constant `SUCCESS_STATUS` is not being used, remove it.

File: lib/checkPullRequestReview.js [hunk 4]
```diff
@@ -181,52 +213,31 @@ const handleApproval = async (context, pullRequest, review) => {
...
-        ' ' +
-        reviewApprovedSearchString +
-        ' ' +
-        pullRequest.number,
-    })
+  // Check if the PR has pending reviews and assign remaining reviewers.
+  // If the PR does not have any pending review, check if it has changes
+  // requested and do nothing since the PR author was already assigned when
+  // the review was made.
+  // If none of the above cases, check if the PR has been approved (the search mechanism)
```
Comment (ankita240796) [inline lib/checkPullRequestReview.js:220]: Explain the search mechanism. Also, add a full stop in end.

File: lib/checkPullRequestReview.js [hunk 5]
```diff
@@ -181,52 +213,31 @@ const handleApproval = async (context, pullRequest, review) => {
...
+  const hasChangesRequested = await utilityModule.doesPullRequestHaveChangesRequested(
```
Comment (ankita240796) [inline lib/checkPullRequestReview.js:221]: Check line lengths - ensure they are less than 81 chars.

**Review by ankita240796 — CHANGES_REQUESTED** [review]
Thanks for the fix @jameesjohn! Just a few nits, PTAL. Thanks!

File: lib/checkPullRequestReview.js (hunk 1 above)
Comment (jameesjohn) [reply lib/checkPullRequestReview.js]: It is not being used. Removed.

File: lib/checkPullRequestReview.js (hunk 4 above)
Comment (jameesjohn) [reply lib/checkPullRequestReview.js]: Done.

File: lib/checkPullRequestReview.js (hunk 5 above)
Comment (jameesjohn) [reply lib/checkPullRequestReview.js]: Done.

File: lib/utils.js (hunk 2 above)
Comment (jameesjohn) [reply lib/utils.js:167]: `SUCCESS_STATUS` is used for checking if the user is a collaborator.

> Explain with a comment why we are checking like this (same as what we discussed offline.)

Done.

File: lib/utils.js (hunk 3 above)
Comment (jameesjohn) [reply lib/utils.js:195]: Done.

**Review by ankita240796 — APPROVED** [review]
LGTM, Thanks @jameesjohn!

**Comment by jameesjohn** [comment]
Thanks, @ankita240796. Please let me know when this gets deployed.

**Pull request merged**